### ISSUE TITLE
Cousins Count Report - Add Smart Sequence Option

### ISSUE THREAD

**Issue by EmmArrBee** [ShammyLevva/FTAnalyzer#140]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
The default sort sequence of the "Cousins Count Report" is based on the text labels in the Relation to Root column.
 
So "cousin" sorts after "aunt" and well away from "1st cousin once removed"

"1st cousin once removed" sorts before "1st cousin twice removed" (that makes sense to me), but both sort before "1st cousin 3 times removed" and "1st cousin 4 times removed"

Can we have a more generational focused sort option, perhaps through the addition of an additional sortable column? If root is g0, then parents might be g-1 and children might be g+1. Siblings might be g0.1 and nephews/nieces g1.1. Aunts/Uncles might be g-1.1

But there may be more program friendly ways of achieving a similar results

**Comment by ShammyLevva** [comment]
I think this is a bit of a holy grail - finding a great way of sorting families logically. I know of no such scheme that would do this sorting.

The point would be that the loading of data would need to create a sort code for each person. Then this sorting order could be applied to any report. However without that "code" to sort on we are stuck. I wonder if the "budgie" code might be utilised. Can you have a play around with the sorting on the main lists that has budgie code field and sort on that and see if that gives closer to what's required?

If not then this might lie open for quite a while until a suitable code is identified that would allow this sort of sorting.

**Comment by EmmArrBee** [comment]
Surely this isn't about sorting families or allocating codes to each person. 

When you load a GEDCOM, you're already dropping those related to the root into buckets you've labelled with terms like "cousin" and "aunt". The request is about being able to sort the labels you've chosen to use in what might be a more meaningful way. It's those labels which need to be sortable.

My main report has 6900+ plus individuals and isn't as focused as this statistical summary

M

**Comment by ShammyLevva** [comment]
Yes so each category needs a sort order so there needs to be a code allocated to everyone based on their relationship type. That means before the sorting is right for each xth cousin there will need to be a method of sorting the perhaps 100 xth cousins. So it’s not just about the right sort order for the relationship type it’s about the right sort order for the lower levels too.

**Comment by ShammyLevva** [comment]
So not just first order sorting - and I’ve no idea how you’d sort them but it’s second and third order sorting too.
